src/auth/auth-cross-domain-iframe.spec.ts

```diff
@@ -0,0 +1,79 @@
+import { BBAuthDomUtility } from '../shared/dom-utility';
+import { BBAuthCrossDomainIframe } from './auth-cross-domain-iframe';
+
+function bindEvent(element: any, eventName: string, eventHandler: any) {
+  if (element.addEventListener) {
+      element.addEventListener(eventName, eventHandler, false);
+  } else if (element.attachEvent) {
+      element.attachEvent('on' + eventName, eventHandler);
+  }
+}
+
+function IFrameMock(frame: HTMLIFrameElement) {
+  bindEvent(frame.contentWindow, 'message', (msg: any) => {
+    if (msg.data['methodName'] === 'ready') {
+      window.postMessage({methodName: 'ready'}, '*');
+    } else if (msg.data['methodName'] === 'getToken') {
+      getTokenCalls += 1;
+      window.postMessage({
+        methodName: 'getToken',
+        value: 'accessToken!'
+      }, '*');
+    }
+  });
+}
+
+let getTokenCalls: number;
+
+describe('Auth Cross Domain Iframe', () => {
+  let fakeIframe: HTMLIFrameElement;
+
+  beforeEach(() => {
+    fakeIframe = document.createElement('iframe');
+    getTokenCalls = 0;
+  });
+
+  it('creates a new frame if none exist', () => {
+    const getElementSpy = spyOn(document, 'getElementById').and.callThrough();
+    const requestSpy = spyOn(BBAuthDomUtility, 'addIframe');
```

Comment:
There's a general lack whitespace in between some lines.  I prefer to see a blank line after `const`/`let` statements unless they're followed by a series of property assignments for constant/variable.  For example, I would change this code to this:

```
    const getElementSpy = spyOn(document, 'getElementById').and.callThrough();
    const requestSpy = spyOn(BBAuthDomUtility, 'addIframe');

    BBAuthCrossDomainIframe.getOrMakeIframe();

    expect(getElementSpy).toHaveBeenCalledWith('auth-cross-domain-iframe');
    expect(requestSpy).toHaveBeenCalledWith('', 'auth-cross-domain-iframe', '');
```